irohad/ametsuchi/tx_cache_outcome.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TX_CACHE_OUTCOME_HPP
+#define IROHA_TX_CACHE_OUTCOME_HPP
+
+#include <functional>
+
+#include <boost/variant.hpp>
+#include <cryptography/hash.hpp>
+
+#include "cryptography/hash.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+
+    /**
+     * Class represents the outcome of tx presence in a cache
+     */
+    class TxCacheOutcome {
```

Comment:
I suggest a bit better wording (please pick any name you like).

The word "outcome" does not suit the context. (discussed that offline)

```suggestion
    class TxCacheResponse {
```


There are a couple of links referring to the problem: [1](https://ell.stackexchange.com/questions/132333/what-is-the-difference-between-result-and-outcome-in-that-context) & [2](https://www.quora.com/What-is-the-difference-between-“result”-and-“outcome”-in-English)